src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -617,6 +617,98 @@ public function testDumpDateTime($dateTime, $expected)
         $this->assertSame($expected, Inline::dump($dateTime));
     }
 
+    /**
+     * @dataProvider getNumericKeyData
+     */
+    public function testDumpNumericKeyAsString(array|int $input, int $flags, string $expected)
+    {
+        $this->assertSame($expected, Inline::dump($input, $flags));
+    }
+
+    public function getNumericKeyData()
+    {
+        yield 'Int with flag' => [
+            200,
+            Yaml::DUMP_NUMERIC_KEY_AS_STRING,
+            '200',
```

Comment:
This does not look expected to me as the int is no key in this context.

Reply:
wrong behavior is `"'200'"`, I thinks is ok here. (changes in Inline class are only in `dumpHashArray` method)

Reply:
yeah, you are right 👍